_ext/events.rb

```diff
@@ -0,0 +1,51 @@
+require 'aweplug/helpers/searchisko'
+require 'parallel'
+
+module JBoss
+  module Developer
+    class Events
+
+      def initialize
+
+      end
+
+      def execute site
+        searchisko = Aweplug::Helpers::Searchisko.default site, 0
+
+        Parallel.each(site.events, in_threads: (site.build_threads || 0)) do |(id, data)|
+
+        searchisko_hash = data.collect { |(key, value)|
+
+          case key
+            when 'title'
+              ['sys_title', value]
+            when 'description'
+              ['sys_description', value]
+            when 'type'
+              ['sys_content_provider', value]
+            else
+              [key, value]
+          end
+        }.to_h
+
+
+        # Add identifier and other required fields for Searchisko Content Object
+        searchisko_hash.merge!({:sys_content_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')})
+        searchisko_hash.merge!({:sys_content_type => 'jbossorg_event'})
+        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
```

Comment:
Yep, though that does become a bit difficult to read, make sure spacing
helps readability, there are a lot of chained methods and three different
key-value pairs being added here.

On Fri, Jan 30, 2015 at 10:12 AM, Dantheman720 <notifications@github.com>
wrote:

> In _ext/events.rb
> <https://github.com/jboss-developer/www.jboss.org/pull/759#discussion_r23855991>
> :
>
> > +            when 'title'
> > +              ['sys_title', value]
> > +            when 'description'
> > +              ['sys_description', value]
> > +            when 'type'
> > +              ['sys_content_provider', value]
> > +            else
> > +              [key, value]
> > +          end
> > +        }.to_h
> > +
> > +
> > +        # Add identifier and other required fields for Searchisko Content Object
> > +        searchisko_hash.merge!({:sys_content_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')})
> > +        searchisko_hash.merge!({:sys_content_type => 'jbossorg_event'})
> > +        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
>
> Ah, I see what you're saying. Ex: searchisko_hash.merge!({:sys_content_id
> => searchisko_hash['sys_title'].downcase.strip.gsub(' ',
> '-').gsub(/[^\w-]/, '') + "-" +
> searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/,
> ''), :sys_content_type => 'jbossorg_event', :sys_url_view =>
> "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
>
> I was tabbing between and was negligent on the merge....sorry about that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jboss-developer/www.jboss.org/pull/759/files#r23855991>
> .
>



-- 
Jason Porter
http://en.gravatar.com/lightguardjp